### PR TITLE
Modify LineList and spectral rebin (and a bit more)

### PR DESCRIPTION
**Pull request by profxj**
A small set of misc. edits:

- bug fix to recover a few H2 lines
- rebin fix to interpolate the edge pixels
- a few tests

### CODE REVIEW

File: linetools/analysis/voigt.py [hunk 1]
```diff
@@ -176,7 +176,7 @@ def voigt_from_abslines(iwave, line, fwhm=None, ret=['vmodel'],
         if const.c.to('km/s')*dwave/medwave > minb/10.:
             wmin = np.min(iwave.to('AA').value)
             wmax = np.max(iwave.to('AA').value)
-            nsub = int(np.round( (np.log10(wmax)- np.log10(wmin)) / 1.449E-6)) + 1
+            nsub = int(np.round( (np.log10(wmax)- np.log10(wmin)) / 1.449E-6)) + 2
```
Comment (ntejos) [inline linetools/analysis/voigt.py]: Why is this needed to be + 2 now? Could you add a inline comment about it?

File: linetools/tests/test_init_absline.py [hunk 2]
```diff
@@ -26,6 +27,12 @@ def test_mk_absline():
     abslin2 = AbsLine(1334.5323*u.AA, linelist='Strong', use_CACHE=True)
     np.testing.assert_allclose(abslin2.data['Ek'], 74932.62 / u.cm)
 
+    # Init CII 1334 with multiple linelists
+    ism = LineList('ISM')
+    H2 = LineList('H2')
+    abslin2b = AbsLine(1334.5323*u.AA, linelist=[ism,H2])
```
Comment (ntejos) [inline linetools/tests/test_init_absline.py]: can you invert the order of the linelist for the test? i.e. `[H2, ism]`

File: linetools/analysis/voigt.py (hunk 1 above)
Comment (profxj) [reply linetools/analysis/voigt.py]: Somehow +1 didn't cover beyond wmax.  It's not 100% necessary
with the extrapolate in rebin, but on the safe side.  Small comment added.

File: linetools/tests/test_init_absline.py (hunk 2 above)
Comment (profxj) [reply linetools/tests/test_init_absline.py]: done

**Comment by profxj** [comment]
merging
